### PR TITLE
Adding Hero and Featured Content Module pages to design manual

### PR DESCRIPTION
**Pull request by stephanieosan**
### Additions
- Added Hero page
- Added Featured content module page
### Notes
- This PR does not include changes to the overall site navigation, since I assume we want to release all of the new pages at once.
- This PR currently does not contain live-coded examples of either the hero or the featured content module. We'll improve on the page in a different PR later to include coded examples after the Capital Framework PRs get merged.
### Testing

Go to the following links to see the pages in the browser:
http://stephanieosan.github.io/design-manual/ui-toolkit/heroes.html
http://stephanieosan.github.io/design-manual/ui-toolkit/featured-content-module.html
### Screenshots

hero-screenshot
fcm-screenshot

### CODE REVIEW

**Comment by jimmynotjim** [comment]
It looks like the font-size in the hero changes based on screen size, can you document that here so it's clear what sizes they should be.

**Comment by stephanieosan** [comment]
Good catch! Just added that the fonts change to H2 and H4 for heros on mobile.

**Comment by Scotchester** [comment]
@stephanieosan In the front-end code for this, the font size changes at 800px, so should your note on where they change appear in the tablet style section, rather than mobile?

File: ui-toolkit/heroes.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+</div>
+
+<div class="content-50 content-last">
+  <h5 class="repo-list-header">Repository</h5>
+  <ul class="repo-list">
+    <li>
+      <span class="cf-icon cf-icon-github"></span>
+    </li>
+    <li>
+      <a href="https://github.com/cfpb/cf-typography"><h4>cf-something</h4></a>
```
Comment (jimmynotjim) [inline ui-toolkit/heroes.md]: This should be updated to `cf-layout`

**Comment by stephanieosan** [comment]
@Scotchester Alternatively, maybe we could somehow more generally indicate that the type follows the normal standards for breaking at smaller screen sizes. Because we're just following other rules here, and if those rules change (like, if we decide to break the type at a different breakpoint), then this would change accordingly. 

@jimmynotjim Updating now!

**Comment by Scotchester** [comment]
@stephanieosan That makes sense. In which case, currently, your placing of the change at mobile size is correct, and the front-end code should change. I'll comment over there accordingly.

**Comment by jimmynotjim** [comment]
@stephanieosan @Scotchester I'm actually inclined to have the headings drop down early, but utilize the new mixins I wrote for CF-Core. The reason is that the h1 and h3 font sizes are entirely too large at tablet min

screen shot 2015-10-02 at 1 36 29 pm

vs 

screen shot 2015-10-02 at 1 36 44 pm

I have a PR for v1 extending the work @virginiacc did and as you can see, it works really well dropping the heading sizes early

**Comment by jenn-franklin** [comment]
Ah, thanks for showing this @jimmynotjim. I agree that the h2/h4 is working better above. Just so I'm totally sure, V1's plan is to have text outside of the hero/FCM still not drop until mobile size? Are there any cases other than the FCM/hero you're dropping to smaller text at this size?

One other question— Are you planning to left + right crop the image above? Or are you finding that to be difficult/awkward at this size? (Or should I not be paying attention to that now and just looking at the text. :smile_cat:)

**Comment by jimmynotjim** [comment]
@huetingj correct, we're using the default text sizes from Capital Framework for all other headings. I only changed it here because of the same change in the Hero PR.

I'm not sure what you mean by left + right crop the image. Can you explain?

**Comment by jenn-franklin** [comment]
Yah! Cropping the image instead of having it proportionally scale down. Here's an early mockup  of the FCM showing it: 
screen shot 2015-10-05 at 10 27 29 am

I think @virginiacc coded it so that the image is being cropped evenly from both sides (until it reaches mobile).

**Comment by jimmynotjim** [comment]
@huetingj ah sorry, that's correct for the FCM, but this is an example of a Hero, I just didn't have an illustration so I'm using an existing image.

**Comment by stephanieosan** [comment]
Sorry, I might have misspoke earlier, I'm just realizing now what y'all are saying and what's going on. 

So, here's my summary:
1. Virginia's pull request has the H1/H3 breaking to H2/H4 on tablet. 
2. This is different from the capital framework standard.
3. @huetingj, if that's intentional and we want to keep it that way (I defer to you), we should document that it breaks on tablet (not mobile, which is where I mistakenly wrote it breaking).
4. If that's not intentional and we want it to break on mobile, like all other type does outside of heros, then we should change it in virginia's PR.

Let me know, so I can make changes in the design manual page accordingly.

**Comment by jimmynotjim** [comment]
Shoot, @Scotchester just pointed out I transposed the 800px and 600px in my comment above. I meant to say that I recommend sticking with the size change in Virginia's PR as it existed because the h1 size at 600px-700px wide screens is entirely too big. Instead, **just for the Heroes**, it should drop from h1 to h2, and h3 to h4 at 800px. Sorry if I caused any confusion.

**Comment by stephanieosan** [comment]
Okay, I just talked to Virginia and Jenn and I think we're all on the same page! 

We're going to leave Virginia's code as-is because we like that better. (Which means Hero text drops at 800px and Featured Content Module drops at 600px.) I'm going to update the design manual pages to properly document that.

Thanks for helping us sort this out. Fixes coming soon!

**Comment by stephanieosan** [comment]
Okay, updates are in. 
- For heros, H1/H3 become H2/H4 at tablet (800px)
- For fcm, the break happens at mobile (600px). I just tweaked the wording to make the formatting match on both pages.

**Comment by stephanieosan** [comment]
So...Is this ready to be merged?

**Comment by benguhin** [comment]
:+1:  for UI

**Comment by Scotchester** [comment]
I have one minor concern:

Can we eliminate the mention of left and right margins? Those margins belong to the overall layout, not the heroes themselves.

**Comment by Scotchester** [comment]
One more minor thing: For consistency of nomenclature with FEWD standards, can we change references to "top and bottom margin" to "top and bottom padding"?

**Comment by Scotchester** [comment]
Clarification: my comment about left and right margins were specific to the heroes page.

**Comment by Scotchester** [comment]
And my comment about margin vs. padding can apply to left and right margins with in a FCM.

**Comment by stephanieosan** [comment]
Yes, I will make those changes.

**Comment by stephanieosan** [comment]
It is so!

**Comment by Scotchester** [comment]
Thanks! Looks good to me :+1: Do we need a Graphic Design approval before merging?

**Comment by stephanieosan** [comment]
I have no idea what the process is. @benguhin or @nataliafitzgerald do y'all want to review this PR again before it gets merged, or was this already reviewed/approved previously?

**Comment by Scotchester** [comment]
Ben gave a thumbs up a few posts up, but Natalia hasn't commented on this PR specifically. Not sure if the content was vetted elsewhere previously.

**Comment by stephanieosan** [comment]
@nataliafitzgerald Can you give this a :+1: if you already vetted the final hero and fcm content elsewhere? Or do a review of this if you haven't?

**Comment by jimmynotjim** [comment]
@stephanieosan Sorry for the last minute request, but I'm just now working on integrating the FCM into v1. 

First, I realized the category is listed as a tag in this PRs docs but since tags are already something else in the Design Manual, can we change text the text here to say category to avoid confusion?

Second, can you explain why there's a change in left padding depending on left-nav/right-sidebar?

**Comment by nataliafitzgerald** [comment]
@stephanieosan - Sure, reviewing now.

**Comment by nataliafitzgerald** [comment]
##### Hero feedback

For the heroes I have added a number of comments to the Google Doc that can be found here: https://docs.google.com/document/d/1JclBNpTKIMQw_GGBIbda351cuZfaQKbEwsnXW4QdKx0/edit#
- In terms of the in-page layout, the 3 columns of bullets in the desktop section feels overwhelming and hard to take in. Did you try an approach where the heroes are to the right of the bullets? To break it up you could show the first set of bullets to the left of the "Before you claim" section and then the second set to the left of the call to action bullets. 
- In reference to: "Top and bottom padding: 45px (In the case that there is minimal text and/or no CTA, the designer may choose to use 60px padding at the top to make the text appear more centered.)" I have seen instances where the content is so short that this looks very off. Should we also set a minimum for the header and supporting text? We should probably include a character count instead of just the number of lines since content creators won't be seeing this in a layout initially. 
- For tablet and mobile add in what the space should be between the header and the subhead.
- For the sections, make the second section "Style" and then make the subsections "Desktop" "Tablet" "Landscape" This is consistent with what we do elsewhere in the DM.
- Global DM inconsistency: Sometimes we are using the header "Stylistic guidelines" and sometimes we are using the header "Style". As we work to update the DM we should be sure to come up with a consistent wording for our sections that are repeated throughout. I also noticed this with "Use" and "Use cases". Another thing I noticed was the use of title casing in some places, for example here: https://cfpb.github.io/design-manual/ui-toolkit/buttons.html And, the naming of the jump links. In most places these are very top level, "Use" "Style" and so on as shown here: http://stephanieosan.github.io/design-manual/ui-toolkit/expandables.html But, in a lot of the new pages that are being written these jump links have much more detailed names (like in this hero page). @benguhin @Dnpizarro  - How can I go about making updates to these types of global things?

Ok, I'll move on to the FCM now.

**Comment by nataliafitzgerald** [comment]
##### Featured content module feedback

I have made a number of comments in the GoogleDoc that can be found here: https://docs.google.com/document/d/11l7_Ff9NHlzySL4IKBeMONYo3j80jfyXwozADNN-2J8/edit#heading=h.l4w13by47777
- In the jump links you use the words "Use cases" and in the section you use the word "Use". Stay consistent. 
- Should the content under "Use cases" be bulleted like it is in the content document?
- Some of the bullets feel really long. Can they be broken up into more digestible points? If we are going to have long bullet text we will have to look at adding more space between our bullets. Our approach in the previous pages we have built out is better, though. For example: http://cfpb.github.io/design-manual/identity/typography.html. The bullets shouldn't be full sentences unless they have to be.
- It looks like there is an extra space in this bullet list. Is that intentional and, if so, what is it indicating? 
  screen shot 2015-10-14 at 12 40 45 pm
- This section is hard to follow. I suggest making the bullets one column and moving all the imagery to the right of the list, including the "Featured" with the star. 
- At Mobile the text says that "minicons drop off" but the image shows that they don't.
- At Mobile and Tablet, tighten up the spacing between the header and subhead.

**Comment by stephanieosan** [comment]
Okay, we've incorporated some changes. This should be ready to merge, @Scotchester !

**Comment by jenn-franklin** [comment]
@jimmynotjim The change in left padding is an attempt to make the text of the FCM align with the rest of the text on the page. Here is the FCM in the context of the two pages:

screen shot 2015-10-19 at 1 22 56 pm

screen shot 2015-10-19 at 1 23 07 pm

File: ui-toolkit/featured-content-module.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+<div class="content-33 content-first">
+* Headline is H2, appears 30px below category and is 1–2 lines
+* Body copy is 16px, appears 30px underneath headline, is 2 to 4 lines
+* A call to action may be used and appears 30px underneath body copy. If a link, standard link styles apply. If a button, standard primary button styles apply.
+</div>
+
+<div class="content-33 content-first">
+* Category is an 18px black minicon and H4 text, appears 30px from top
+* The following five category categories can be used. If the content does not fit within one of the five category categories, you may use this icon and prefix:
```
Comment (Scotchester) [inline ui-toolkit/featured-content-module.md]: "category categories" appears twice on this line.

**Comment by stephanieosan** [comment]
That is because I did a find and replace of tag for category and did not proof read. Good catch, should be fixed now!

**Comment by Scotchester** [comment]
Woohoo!

**Comment by jimmynotjim** [comment]
@stephanieosan @huetingj I'm working on integrating @virginiacc's work into v1, and I have a few questions. 
1. The height of 270, is that firm? What if there's content that forces it larger? Should we make that a min-height instead? I'm seeing examples where the text is being cut short
   
   screen shot 2015-10-28 at 5 02 37 pm
2. The left padding of 60px seems to be incorrect. The default grid uses 30px of gutter on each side, so it seems this module should match the wells (ex http://beta.consumerfinance.gov/the-bureau/) and won't need a change in padding depending on placement.
   
   screen shot 2015-10-29 at 12 22 05 pm
3. At the min tablet and desktop (soon to be bp-sm-min and bp-med-min) sizes, the image is really limited in width. Should the text block be a static size or should it really be a % size?
   **Tablet Min:**
   screen shot 2015-10-29 at 11 23 03 am
   
   **Desktop Min:**
   screen shot 2015-10-29 at 11 23 55 am

_ignore the border color in these examples, still working things out_

**Comment by jenn-franklin** [comment]
Update: Min height will be set to 270px. Will think about adding a character limit to ensure FCMs aren't written to be too tall. Awesome to 30px padding on the left; this was my misunderstanding of the grid. @jimmynotjim and I talked about putting more relevant images (sorry, cat) in code to better visualize and straighten out the design at bp-sm-min and bp-med-min. More to come soon.

**Comment by jimmynotjim** [comment]
After our conversation today we decided to try adding padding to center the text and image inside the FCM at all breakpoints without having to increase the size of the image.

<= 600px - 30px both sides

**Shown at 700px**
screen shot 2015-11-03 at 5 45 29 pm

601px-700px (extra BP just for this) -  80px both sides

**Shown at 601px**
screen shot 2015-11-03 at 5 47 13 pm

701px-900px - Same as spec for tablet

**Shown at 701px**
screen shot 2015-11-03 at 5 48 11 pm

901px-1230px+ - Same as spec for desktop

**Shown at 901px**
screen shot 2015-11-03 at 5 49 00 pm

The event map works really well so I didn't bother with screenshots, but if there's anything else I can show, let me know.

**Comment by schaferjh** [comment]
@jimmynotjim What's the scaling/crop on the image that happens at 901? At first glance, it looks like that could be a very difficult crop like the one we were trying to avoid at tablet. And forgive my obtuseness, but then as the screen gets wider above 901 does the area of image shown extend somehow? Equally on both sides of the image or in one direction?

**Comment by jenn-franklin** [comment]
Wait— I think maybe the images for 901px and 701px are switched above? 

I think this works well. The cropping does look awkward above, but I think that's where @duelj's idea of having a safe space for the image comes in, and I can add that guidance to the AI file. So, it would take some image conscientiousness on behalf of the designer for sure— or really just the need to have horizontally centered imagery. But at least the image wouldn't also need to fit a very wide and short crop to accommodate the 600-700px range. I think the padding eases that transition well.

**Comment by jimmynotjim** [comment]
@huetingj nope, they're not switched, at 701 device width the hero goes split screen but takes up the entire screen. At 901 the right sidebar comes in (same for the left, but I didn't take a screen shot of that one).

**Comment by jimmynotjim** [comment]
@schaferjh The image itself is16:9 at 540x304 (odd, I know), which is displayed full size under 701px, but what happens is the text block is 360px, including padding in the 701-900 range and 330px at the 901 and up. The image is set to take up the remaining space in the hero, whatever it happens to be, by clipping the left and right side evenly. 

Here's a screen shot at 900px (notice the image isn't large enough, we need to make an adjustment here):

screen shot 2015-11-04 at 4 18 55 pm

And one at 1230px:

screen shot 2015-11-04 at 4 19 09 pm

**Comment by jenn-franklin** [comment]
:+1:

**Comment by jimmynotjim** [comment]
@huetingj what should we do about the image not covering at 900px (first of my two newer screenshots)? We could make the text box 30px wider, to 390px, or we could spec the image to be larger, like 570x320, which could reduce that extra padding a bit at the 601-700 range.

**Comment by jenn-franklin** [comment]
@jimmynotjim I vote for increasing the image size

**Comment by jimmynotjim** [comment]
Done! I'll send these changes to @virginiacc's PR so she can give it a test.

**Comment by jenn-franklin** [comment]
Thank you so much for all of your work! It looks great.

**Comment by jimmynotjim** [comment]
Oh, one thing I'm noticing is that we'd need to make the min-height in the 701-900 range 320. Otherwise it resizes the image down to fit the height and a gap continues. Are you ok with that? Here's a screenshot

screen shot 2015-11-05 at 2 03 16 pm

**Comment by jenn-franklin** [comment]
Yah totally--- I actually like it better because it's more accommodating text-wise as a default.

**Comment by jimmynotjim** [comment]
Awesome, thanks for the quick feedback. Going to button this up now.

**Comment by jimmynotjim** [comment]
PR opened https://github.com/virginiacc/cf-layout/pull/2
